assemblyline_v4_service/common/dynamic_service_helper.py

```diff
@@ -2495,6 +2500,18 @@ def _convert_events_dict_to_tree(
         :param events_dict: A dictionary of events
         :return: A list of event tree roots, each which their respective branches and leaves
         """
+        def _depth(d: Dict[str, Any]) -> int:
```

Comment:
Could you add a test with the expected value for those two dicts?

`{
    "children": [
        {"children": [{"children": "A"}]},
        {"children": "A"},
    ],
}`

`{
    "children": [
        {"children": "A"},
        {"children": [{"children": "A"}]},
    ],
}`